Improving read_file function

rhydham-nith opened the pull request:
I've tried to address the issue #997 of `read_file`  for reading a .shp that doesn't have a `image_path` column. 

### Changes Made:

1. Added `image_path` and `label` as optional arguments in `read_file`.
2. Improved type handling using `isinstance()` for better input validation.

**Question:** Should I try to update the documentation about how to use the `read_file` Function

I would be happy to contribute more to the project and am open to any feedback or suggestions for improvement. Thank you for the opportunity to collaborate!

rhydham-nith commented:
Hi @bw4sz  @ethanwhite @henrykironde ,
This PR has passed all the Continuous Integration tests successfully and is ready for review. Whenever you get a chance, could you please take a look and provide me with your feedback? Looking forward to your thoughts!

Thanks in advance!

bw4sz reviewed (changes requested):
Thanks for your contribution, great start. 

1. Update the docstrings
2. Make it clear in both docstrings and as a warning to users that by passing a label or image_path column, that is being applied to the entire dataframe. For example, if I had a dataframe containing annotations from 10 different images, I wouldn't want to pass the image_path argument, this is only in the case of a single image's worth of annotations.
3. Update deepforest docs to show this behavior, with the same text https://github.com/weecology/DeepForest/blob/cda5a907d6a349c6074689e57ad69389ba415b9c/docs/user_guide/01_Reading_data.md?plain=1#L188

rhydham-nith commented:
Sir, 
I’ve made all the desired updates — updated docstring, added example usage, warning, and updated the `.md` file accordingly. Please let me know the next steps. 
Thank you
